src/Tmds.Ssh/PrivateKeyCredential.cs

```diff
@@ -2,15 +2,18 @@
 // See file LICENSE for full license details.
 
 using System;
+using System.Text;
 
 namespace Tmds.Ssh;
 
 public sealed class PrivateKeyCredential : Credential
 {
     internal string FilePath { get; }
+    internal byte[] Passphrase { get; }
 
-    public PrivateKeyCredential(string path)
+    public PrivateKeyCredential(string path, string? passphrase = null)
     {
         FilePath = path ?? throw new ArgumentNullException(nameof(path));
+        Passphrase = string.IsNullOrEmpty(passphrase) ? Array.Empty<byte>() : Encoding.UTF8.GetBytes(passphrase);
```

Comment:
Can we treat the empty non-null string as a password that is empty (instead of it to mean the same thing as `null`)?

Reply:
... because though it isn't a good idea to use an empty string as a password, I assume it is possible.

Reply:
It may be possible with `openssl` to generate a PKCS#1 key but `ssh-keygen` treats `-N ''` to be no password. It probably is possible to build the encryption key from an empty string manually but not using the normal tools. Because of this I thought treating an empty string the same as `null`/no password. It was also done because the underlying method to decrypt the code was using a `ReadOnlySpan<byte>` so was easier to treat an empty span as no password set.

Reply:
Ok, let's keep it as is.

Reply:
I'd like to come back on this. I think the .NET semantics should be null for no password and not null for a password.

Next to the Span, we can add a bool that tracks if a password was provided.

Reply:
Sure we can do that

Reply:
Can you also add a `Func<string?> prompt)` overload, like we have for `PasswordCredential`?